### PR TITLE
Update upgrade doc and make generic

### PR DESCRIPTION
**Pull request by willingc**
Partially addresses #392.

- Reordered some sections starting with tips and warnings
- Shorten specific release sections
    - refer 0.6 upgrade to CHANGELOG
    - refer 0.5 upgrade to 0.5 docs and remove 0.5 specific information
- Make subtopics/subsystems generic
- Add a very brief troubleshooting tip for upgrading

### CODE REVIEW

**Comment by choldgraf** [comment]
This looks great! Made some very small type-o edits and added an extra link in there, thanks @willingc !

File: doc/source/upgrading.md [hunk 1]
```diff
@@ -1,33 +1,78 @@
...
-In general, it should be safe to upgrade a minor revision of the chart to
-receive bugfixes and improvements. A major chart upgrade (e.g. v0.4 to v0.5)
-can be more challenging, but is possible. We still recommend a full fresh
-installation if you can, but recognize it is not possible all the time. We've
-provided and tested the following upgrade instructions. If you are planning an
-upgrade of a critical major installation, we recommend you test it out on a
-staging cluster first before applying it to production. Feel free to reach out
-to us on [gitter](http://gitter.im/jupyterhub/jupyterhub) or the [mailing
-list](https://groups.google.com/forum/#!forum/jupyter) for upgrade help!
+- We **recommend** a full fresh installation instead of upgrade if you can,
```
Comment (yuvipanda) [inline doc/source/upgrading.md]: I don't think we recommend this in general, only for specific cases (such as for 0.4 - 0.5). I don't think we should mark this as a general recommendation.

File: doc/source/upgrading.md [hunk 2]
```diff
@@ -1,33 +1,78 @@
...
-provided and tested the following upgrade instructions. If you are planning an
-upgrade of a critical major installation, we recommend you test it out on a
-staging cluster first before applying it to production. Feel free to reach out
-to us on [gitter](http://gitter.im/jupyterhub/jupyterhub) or the [mailing
-list](https://groups.google.com/forum/#!forum/jupyter) for upgrade help!
+- We **recommend** a full fresh installation instead of upgrade if you can, 
+  but recognize that it is not possible all the time. 
+- Minor revisions: It should be safe to upgrade a minor revision of the chart to
+  receive bugfixes and improvements.
+- Major chart upgrades (e.g. v0.5 to 0.6): This is more challenging than a
```
Comment (yuvipanda) [inline doc/source/upgrading.md]: We also try to make these as easy as possible, and put a bit of effort into it. I don't think we should call this challenging.

File: doc/source/upgrading.md (hunk 1 above)
Comment (yuvipanda) [reply doc/source/upgrading.md]: I think I'll actually just send a followup PR instead of nitpicking here! Didn't realize it got merged :)

**Comment by choldgraf** [comment]
(I can spot-check @yuvipanda comments in the upcoming changelog PR if that works for you all)

File: doc/source/upgrading.md (hunk 1 above)
Comment (willingc) [reply doc/source/upgrading.md]: Makes sense.

File: doc/source/upgrading.md (hunk 2 above)
Comment (willingc) [reply doc/source/upgrading.md]: Lol. That was left over wording from before.. Please strike.

**Pull request merged**